src/libs/SidebarUtils.ts

```diff
@@ -587,21 +587,32 @@ function getRoomWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
         welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfArchivedRoomPartOne');
         welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfArchivedRoomPartTwo');
     } else if (ReportUtils.isDomainRoom(report)) {
+        welcomeMessage.showReportName = false;
         welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryDomainRoomPartOne', {domainRoom: report?.reportName ?? ''});
         welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryDomainRoomPartTwo');
     } else if (ReportUtils.isAdminRoom(report)) {
+        welcomeMessage.showReportName = false;
         welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryAdminRoomPartOne', {workspaceName});
         welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryAdminRoomPartTwo');
     } else if (ReportUtils.isAnnounceRoom(report)) {
+        welcomeMessage.showReportName = false;
         welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryAnnounceRoomPartOne', {workspaceName});
-        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryAnnounceRoomPartTwo', {workspaceName});
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryAnnounceRoomPartTwo');
     } else if (ReportUtils.isInvoiceRoom(report)) {
         welcomeMessage.showReportName = false;
-        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryInvoiceRoom');
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryInvoiceRoomPartOne');
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryInvoiceRoomPartTwo');
+        const payer =
+            report?.invoiceReceiver?.type === CONST.REPORT.INVOICE_RECEIVER_TYPE.INDIVIDUAL
+                ? ReportUtils.getDisplayNameForParticipant(report?.invoiceReceiver?.accountID)
+                : PolicyUtils.getPolicy(report?.invoiceReceiver?.policyID)?.name;
+        const receiver = ReportUtils.getPolicyName(report);
+        welcomeMessage.messageText = `${welcomeMessage.phrase1}${payer} ${Localize.translateLocal('common.and')} ${receiver}${welcomeMessage.phrase2}`;
+        return welcomeMessage;
     } else {
         // Message for user created rooms or other room types.
         welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryUserRoomPartOne');
-        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryUserRoomPartTwo');
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryUserRoomPartTwo', {displayName: ReportUtils.getPolicyName(report)});
```

Comment:
On this line using `ReportUtils.getPolicyName(report)` for the `displayName` does not yield the correct result because for WS #rooms it will say that the room was created by the {workspaceName} when it should be the name of the user that created the room aka the Workspace owner.

The reason I was able to point this out was because the copy says:
> This chat room is for anything #[roomName] related. It was created by [displayName].

And this is how the workspace room looks on my side, it should say `Zerax` instead of `ZeraWS`:

<img width="972" alt="Screenshot 2024-10-02 at 17 03 31" src="https://github.com/user-attachments/assets/84585705-6f0d-41a3-b835-fb712ab7f49d">

Reply:
The problem here is in the report data we cannot know who is the user that created the room.

Reply:
@nkdengineer Cool, thanks for the context. I'll bring this up with the team in the issue to see whether we leave it as is or change it.

♻️ Will proceed with the checklist  in the meantime.

Reply:
@nkdengineer Looks like out of the options provided by me in https://github.com/Expensify/App/issues/47427#issuecomment-2390886282, the [decision](https://github.com/Expensify/App/issues/47427#issuecomment-2391839528) was to simply remove the `It was created by [displayName].` part from the #room copy.

♻️ I will move forward with completing the checklist today as everything else looks good, and whenever you get a chance please remove that part. Thank you!

Reply:
@ikevin127 I updated.